src/DataIntegrity.php

```diff
@@ -105,6 +108,38 @@ public static function ensureFieldsPresent(dict<string, mixed> $row, table_schem
               } else {
                 $row[$field_name] = (int)$row[$field_name];
               }
+            } else if ($row[$field_name] is int) {
+
+                $row[$field_name] = (int)$row[$field_name];
+                $signed = !($field_unsigned);
+
+                switch($field_mysql_type) {
+                  case DataType::TINYINT:
+                    if ($row[$field_name] >= (($signed) ? -\pow(2,7) : 0) && $row[$field_name] < (($signed) ? \pow(2,7) : \pow(2,8))){
+                      break;
+                    }
```

Comment:
I think` // FALLTHROUGH` will make this more inefficient since it will check every case before throwing. What about inverting the conditions here and throwing inside the if, otherwise break outside the if?

The math seems correct to me on all of these.